src/models/output.ts

```diff
@@ -210,6 +233,44 @@ class Output {
 
     return [output, outputBuffer];
   }
+
+  /**
+   * Checks if the script length is within the valid limits
+   *
+   * @returns {boolean} True if the script is within valid limits
+   *
+   * @memberof Output
+   * @inner
+   */
+  hasValidLength(): boolean {
+    // No script can have more than the maximum length
+    if (this.script.length > MAXIMUM_SCRIPT_LENGTH) {
+      return false;
+    }
+
+    // P2PKH and P2SH scripts can only have 30 or 25 bytes (with or without timelock)
+    if (P2PKH.identify(this.script) || P2SH.identify(this.script)) {
+      if (!(this.script.length === 25 || this.script.length === 30)) {
+        return false;
+      }
+    }
+
+    return true;
```

Comment:
```suggestion
    return P2PKH.identify(this.script) || P2SH.identify(this.script)
```

This would include all cases, like @pedroferreira1 said on this [comment](https://github.com/HathorNetwork/hathor-wallet-lib/pull/378/files#r917592898)